tests/system/Debugger/E2ETest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Tests\System\Debugger;
+
+use Google\Cloud\Debugger\V2\Gapic\Debugger2GapicClient as GapicClient;
+use Google\Cloud\Debugger\V2\Breakpoint;
+use Google\Cloud\Debugger\V2\SourceLocation;
+use Google\Cloud\TestUtils\EventuallyConsistentTestTrait;
+use Google\Cloud\TestUtils\AppEngineDeploymentTrait;
+use GuzzleHttp\Client;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * The test deploys the sample app contained in the app folder to Google App
+ * Engine Flexible Environment. Before deployment, we generate a composer.json
+ * that uses the current branch of google/cloud to test against.
+ *
+ * Each test sets a breakpoint, ensures that the app has seen the breakpoint,
+ * then makes a request to the app that should trigger the breakpoint. We then
+ * ensure that the breakpoint has been fulfilled.
+ *
+ * @group debugger
+ */
+class E2ETest extends TestCase
+{
+    protected static $debuggeeId;
+    protected static $httpClient;
+
+    use AppEngineDeploymentTrait;
+    use EventuallyConsistentTestTrait;
+
+    public static function beforeDeploy()
+    {
+        self::createComposerJson();
+        self::$gcloudWrapper->setDir(implode(DIRECTORY_SEPARATOR, [__DIR__, 'app']));
+    }
+
+    public static function afterDeploy()
+    {
+        $url = self::$gcloudWrapper->getBaseUrl();
+        self::$httpClient = new Client(['base_uri' => $url]);
+
+        $resp = self::$httpClient->get('/debuggee');
+        $data = json_decode($resp->getBody()->getContents(), true);
+        self::$debuggeeId = $data['debuggeeId'];
+    }
+
+    public static function tearDownAfterClass()
+    {
+        self::deleteApp();
+    }
+
+    public static function createComposerJson()
+    {
+        $branch = exec('git rev-parse --abbrev-ref HEAD');
+        $origin = exec('git remote get-url origin');
```

Comment:
WDYT about requiring the remote name be set as an environment variable? I'm not sure I love that solution, but expecting the remote name to be origin seems fragile and it may not always be correct.

Reply:
Adding the ability to specify the repo slug (available on travis) as `[user]/[repo]`. It will default to `GoogleCloudPlatform/google-cloud-php`